### DISCUSSION TITLE
support COMMENT ON EXTENSION

### DISCUSSION THREAD

**Discussion by gregwebs** [sqldef/sqldef#620] [Feature requests]
## Overview

`COMMENT ON` is supported in #266, but not for `EXTENSION`.

I wouldn't mind if the comments end up getting dropped, just as long as the SQL gets parsed- in that case a workaround is to filter the comment out of the schema.sql before it gets to sqldef.

```
psqldef ... <<< $(grep -v 'COMMENT ON EXTENSION' schema.sql)
```

## Platform

* OS: macOS (probably all)
* RDBMS: PostgreSQL
* Version: v0.15.12

## Input SQL

```
CREATE EXTENSION IF NOT EXISTS pg_stat_statements WITH SCHEMA public;
COMMENT ON EXTENSION pg_stat_statements IS 'track execution statistics of all SQL statements executed';

CREATE EXTENSION IF NOT EXISTS pgcrypto WITH SCHEMA public;
COMMENT ON EXTENSION pgcrypto IS 'cryptographic functions';
```

## Current output

```
found syntax error when parsing DDL "COMMENT ON EXTENSION ...
```

## Expected output

None compared to the above

**Comment by k0kubun** [comment]
Thanks for filing an issue. Would you be interested in working on a pull request for it? It would look similar to what https://github.com/k0kubun/sqldef/commit/b03a6e4ea16c1b6ca2cc3fe0b0e86a8ef8fad0ad is doing.

**Comment by gregwebs** [comment]
It's not clear to me from that change where the parser is actually failing and would need to be modified. I am content to just use my workaround since none of the extension comments I see seem useful.
